src/tree.rs

```diff
@@ -95,6 +95,16 @@ impl From<(incrementalmerkletree::Position, Vec<MerkleHashOrchard>)> for MerkleP
     }
 }
 
+impl From<incrementalmerkletree::MerklePath<MerkleHashOrchard, 32>> for MerklePath {
+    fn from(path: incrementalmerkletree::MerklePath<MerkleHashOrchard, 32>) -> Self {
+        let position: u64 = path.position().into();
+        Self {
+            position: position as u32,
+            auth_path: path.path_elems().try_into().unwrap(),
```

Comment:
Why does `incrementalmerkletree::MerklePath` store the values as a `Vec` (which is what necessitates returning a slice from `incrementalmerkletree::MerklePath::path_elems`), when `incrementalmerkletree::MerklePath` has a generic parameter for the tree depth (and thus the length of the path) and could store as an array?

Reply:
Historic reasons; `incrementalmerkletree::MerklePath` was brought over directly from `librustzcash` and then had the `DEPTH` parameter added. `DEPTH` is a `const u8` (for better compatibility with `Level`) and I tried today to bring it into line with the `Orchard` design but I couldn't figure out how to use `DEPTH` in defining the length of the array, which requires it to be a `usize`.